projects/lib/src/oauth-service.ts

```diff
@@ -1083,12 +1083,16 @@ export class OAuthService extends AuthConfig {
                 );
             }
 
-            if (this.oidc && this.requestAccessToken) {
-                this.responseType = 'id_token token';
-            } else if (this.oidc && !this.requestAccessToken) {
-                this.responseType = 'id_token';
+            if (this.config.responseType) {
+              this.responseType = this.config.responseType;
             } else {
-                this.responseType = 'token';
+              if (this.oidc && this.requestAccessToken) {
```

Comment:
could be else if, too, right? That would simplify this if/else/else if complexity, I think :)